File: single-number/single-number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        solution = {}
        
        for i in nums:
            if i in solution:
                solution[i] += 1
            else:
                solution[i] = 1
        
        for key, value in solution.items():
            if value == 1:
                return key 

File: single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        if (nums.length == 1) {
            return nums[0];
        } else {
            int unique = 0;
            
            for (int i = 0; i < nums.length; i++) {
                unique ^= nums[i];
            }
            
            return unique;
        }
        
    }
}